### PR TITLE
add semaphore test

### PR DESCRIPTION
**Pull request by mehlis**

### CODE REVIEW

**Comment by mehlis** [comment]
merge together with https://github.com/RIOT-OS/RIOT/pull/218

**Comment by OlegHahm** [comment]
There is no err.h for newlib and msp430lib.

File: test_semaphore/main.c [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+    puts("first: sem_destroy");
+    if (sem_destroy(&s) != 0) {
+        err(1, "first: sem_destroy failed");
+    }
+
+    puts("first: end");
+}
+
+static void priority_sema_thread()
```
Comment (OlegHahm) [inline test_semaphore/main.c]: void parameter is missing.

File: test_semaphore/main.c [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+{
+    sem_wait(&s);
+    printf("Thread '%s' woke up.\n", thread_getname(thread_getpid()));
+}
+
+char names[N][16];
+void test2()
```
Comment (OlegHahm) [inline test_semaphore/main.c]: void parameter is missing.

**Comment by mehlis** [comment]
1. removed err.h dependency
2. added void as function parameter

**Comment by mehlis** [comment]
this PR depends on https://github.com/RIOT-OS/RIOT/pull/246

File: test_semaphore/main.c [hunk 3]
```diff
@@ -0,0 +1,124 @@
+#include <stdio.h>
+
+#include "thread.h"
+#include "semaphore.h"
+
+#define N 10
```
Comment (OlegHahm) [inline test_semaphore/main.c]: Stupidly enough but:

```
main.c:6:0: warning: "N" redefined [enabled by default]
/usr/lib/gcc/msp430/4.6.3/../../../../msp430/include/msp430f1612.h:97:0: note: this is the location of the previous definition
```

**Comment by OlegHahm** [comment]
MSB-430H reboots after

```
26  +    puts("second: end"); 
```

**Comment by OlegHahm** [comment]
Ok, I think the problem is that thread termination seems unstable on the MSP430. Putting a `thread_sleep()` at the end of `test1_second_thread()` and `priority_sema_thread()` helps.
However, there are not enough PRIORITIES on 16-bit platforms, so that the creation of the 6th thread fails.

**Comment by OlegHahm** [comment]
Still crashes on the MSB-430 without the `thread_sleep()` at the end of each thread function, but that seems to be bug in the the MSP430 port. ACK.

**Pull request merged**